securedrop/models.py

```diff
@@ -446,7 +452,7 @@ class Journalist(db.Model):
 
     def __init__(self,
                  username: str,
-                 password: str,
+                 password: 'Optional[str]',
```

Comment:
I think this make the Journalist class' API less clear: the reader might assume that for any journalist, the password is optional. My recommendation is to leave the password as a non-optional str, and move the logic to handle the special case of the deleted journalist somewhere else (so that the special case doesn't "pollute" the "general" API).

Reply:
Good point. I think we can have DeletedJournalist just pass in `password='unused'` since set_password() is a no-op. And the string "unused" is too short, so if it somehow did make it into a real password field it would get rejected.